roles/variables/defaults/main.yml

```diff
@@ -2555,4 +2555,5 @@ ccloud_schema_registry_key: xxxxx
 ### CCloud Schema Registry Secret
 ccloud_schema_registry_secret: yyyyyy
 
-kafka_sasl_enabled_mechanisms: "{{ (kafka_broker_sasl_enabled_mechanisms + kafka_controller_sasl_enabled_mechanisms) | unique }}"
+kafka_sasl_enabled_mechanisms: >-
+  {{ (kafka_broker_sasl_enabled_mechanisms + kafka_controller_sasl_enabled_mechanisms) | unique if kraft_enabled|bool else kafka_broker_sasl_enabled_mechanisms }}
```

Comment:
what was the need of the kraft enabled condition here though? won't it automatically ignore controller sasl mechanisms in zk setup?

Reply:
It would automatically ignore but wanted to add it from clarity perspective from code.